lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -1281,6 +1312,8 @@ using namespace Js;
         case PropertyIds::source:
         case PropertyIds::options:
             SET_PROPERTY(!scriptConfig->IsES6RegExPrototypePropertiesEnabled());
+        case PropertyIds::dotAll:
+            SET_PROPERTY(scriptConfig->IsES9RegExDotAllEnabled() && !scriptConfig->IsES6RegExPrototypePropertiesEnabled());
```

Comment:
>ES9 [](start = 41, length = 3)

does this ES2018 feature check make sense with the ES6 check?

Reply:
Yeah I guess this would have to be blocked on the same feature.

---
In reply to: [210786581](https://github.com/Microsoft/ChakraCore/pull/5592#discussion_r210786581) [](ancestors = 210786581)

Reply:
It's not blocked on the ES6 feature check it's rather that all of these properties behave differently without the ES6 Prototype properties enabled.

With ES2018RegExDotAllEnabled and ES6RegExPrototypePropertiesEnabled the prototype has a dotAll getter only property which returns if the RegEx it's called on has the dotAll flag.

If dotAll is enabled but the Prototype properties are not enabled it's a property directly on the RegExp object which returns true (as opposed to a getter from the prototype)

Reply:
Right, I had a look and found the same. Thanks for clarifying.